src/components/shop/menu.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @module components/shop
+ * @copyright BBC 2020
+ * @author BBC Children's D+E
+ * @license Apache-2.0
+ */
+import { getSafeArea } from "./shop-layout.js";
+import { createGelButtons, resizeGelButtons } from "./menu-buttons.js";
+
+export const createMenu = (scene, config) => {
+    const bounds = getSafeArea(scene.layout);
+    const { buttonsRight } = config;
+
+    const menuContainer = scene.add.container();
+    menuContainer.config = config;
+    menuContainer.setVisible = setVisible(menuContainer);
+    menuContainer.resize = resize(menuContainer);
+    menuContainer.memoisedBounds = bounds; // gel button offset bug means we can't rely on getBounds() for now
+
+    menuContainer.add(createRect(scene, bounds, buttonsRight));
+    menuContainer.add(createButtonRect(scene, bounds, !buttonsRight));
+    menuContainer.add(createInnerRect(scene, bounds, buttonsRight));
+    const yOffset = bounds.height / 2 + bounds.y;
+    menuContainer.buttons = createGelButtons(scene, getInnerRectBounds(bounds, buttonsRight), config, yOffset);
+
+    menuContainer.setY(yOffset);
+    return menuContainer;
+};
+
+const createButtonRect = (scene, menuBounds, isOnLeft) => {
+    const { x, y, width, height } = getHalfRectBounds(menuBounds, isOnLeft);
+    return scene.add.rectangle(x, y, width, height, 0xff00ff, 0.3);
+};
+
+const createInnerRect = (scene, outerBounds, buttonsRight) => {
+    const isOnLeft = !buttonsRight;
+    const bounds = getInnerRectBounds(outerBounds, isOnLeft);
+    return scene.add.rectangle(bounds.x, bounds.y, bounds.width, bounds.height, 0x0000ff, 0.3);
+};
+
+const getInnerRectBounds = (outerBounds, isOnLeft) => {
```

Comment:
looks similar to getHalfRectBounds, wonder if any refactoring could reduce this to 1 function - maybe not doable without making the function messier

Reply:
The two bounds functions are a bit different mathematically but I _was_ able to squeeze a bit more code reuse out of this by generalising the various funcs that created rects. 3e962d0